frontend/src/components/DockerDesktopSetup/DockerDesktopSetup.stories.tsx

```diff
@@ -0,0 +1,25 @@
+import { Meta, Story } from '@storybook/react/types-6-0';
+import { Provider } from 'react-redux';
+import defaultStore from '../../redux/stores/store';
+import DockerDesktopSetup from './DockerDesktopSetup';
+
+export default {
+  title: 'DockerDesktopSetup',
+  component: DockerDesktopSetup,
+  argTypes: {},
+  decorators: [
+    Story => {
+      return (
+        <Provider store={defaultStore}>
+          <Story />
+        </Provider>
+      );
+    },
+  ],
+} as Meta;
+
+const Template: Story = args => {
+  return <DockerDesktopSetup refreshConfig={false} {...args} />;
+};
+
+export const DockerDesktopDefault = Template.bind({});
```

Comment:
Having the different states in here means they can be tested (both for regressions with storyshot snapshot testing, and with the UI).

> The idea is that you do all the messy “connected” logic in a single wrapper component in your app outside of Storybook
https://storybook.js.org/docs/react/writing-stories/build-pages-with-storybook#pure-presentational-pages

How to split this into a presentational component and a container component? All the state/effect/api calls/handlers would happen in a DockerDesktopSetup component. The DockerDesktopSetupPure component would take props for those state, only presentational parts. DockerDesktopSetup would pass all of these into DockerDesktopSetupPure. Then the storybook would test the DockerDesktopSetupPure component. Finally, you shouldn't have to have any mocks in there to test DockerDesktopSetupPure.

Reply:
Created a pure component and updated the storybook 👍🏽